TDAT: public void test looping0 ( ) throws exception {  <NL> list list = new array list ( ) ;  <NL> looping iterator loop = new looping iterator ( list ) ;  <NL> assert true ( " has next should return false " , loop . has next ( ) = = false ) ;  <NL>  <NL> try {  <NL> loop . next ( ) ;  <NL> fail ( " no such element exception was not thrown during next ( ) call . " ) ;  <NL>  } catch ( no such element exception ex ) {  <NL>  }  <NL>  }  COM: <s> tests whether an empty looping iterator works as designed </s>